web/src/components/vis/mixins/axisPlot.js

```diff
@@ -86,16 +109,15 @@ export const axisPlot = {
       }
 
       // Label the axes.
-      this.setXLabel(xlabel);
-      this.setYLabel(ylabel);
+      this.setXLabel(svg, xlabel);
+      this.setYLabel(svg, ylabel);
 
       // Conclude the initialization process.
       this.axisPlotInitialized = true;
     },
 
-    setXLabel(msg) {
+    setXLabel(svg, msg) {
```

Comment:
Would it be bad to "capture" `svg` during `axisPlot` as local data, then use it here, instead of forcing the caller to supply it? This is coming from my obvious background having grown up as an OOP programmer and wanting to encapsulate this bit of data, since it is not supposed to change over the lifetime of this mixin's instantiation.

Reply:
axisPlot is not a constructor, so it can't be guaranteed that it will be called before other methods.   For that reason, I'm hesitant to treat it like a constructor.  We don't really have a paradigm for communicating that.

I also don't see why this mixin couldn't be used where the instantiation component managed a couple of plots.  For example, two plots where the domain and range of the data will be identical.

Reply:
But can't it be guaranteed that a given user of this mixin can arrange to explicitly call `axisPlot()` before ever calling e.g. `setXLabel()`?

I like the idea of getting the mixin to do double duty, but it gets dicey because `margin` for instance doesn't vary, while the target SVG element would (and presumably the two plots you're managing could in principle have different margin values). My feeling is that if we want to be able to drive two plots within a single component, we'd want to design this mixin to explicitly be able to do that, rather than just pass an SVG element to the utility methods without covering the other aspects of plot management.

Reply:
(also discussed this in person; decided to let `axisPlot` capture the SVG element to avoid having to leak the SVG element in calls to the set label functions.)